### PR TITLE
FIX-#2976: Update pandas to 1.2.4

### LINKED ISSUE #2976: Update pandas to 1.2.4

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2976  
- [x] tests passing

### CODE REVIEW

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Review by devin-petersohn — DISMISSED** [review]
Thanks @YarShev!

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
@YarShev Can we use `xfail` instead of `skipif` in the parameter markers?

**Comment by vnlitvinov** [comment]
> @YarShev Can we use `xfail` instead of `skipif` in the parameter markers?

I know the original author is going to be away for a few days, so I decided to step in and do the requested to speed up the things. I hope it's fine for everyone 😄

**Comment by devin-petersohn** [comment]
@vnlitvinov It is ok with me, but we have some failures because `xfail` should always fail in our tests.

It looks like there are conditions where it still works.

**Comment by vnlitvinov** [comment]
For reference: link to relevant Pandas issue - https://github.com/pandas-dev/pandas/issues/41074

I would suggest waiting for @YarShev to step up then, as I cannot locally reproduce the issues he was marking as `skipif`/`xfail` - if I update `pandas` to 1.2.4, take his branch and remove this `xfail` the tests pass... but he did mark those for some reason.

**Comment by YarShev** [comment]
@devin-petersohn , @vnlitvinov , I got rid of xfail/skipif marks in the tests but it had to make some changes in the code to match pandas behavior.

File: modin/backends/base/query_compiler.py [hunk 1]
```diff
@@ -573,21 +573,57 @@ def min(self, **kwargs):
...
+    def sum(self, squeeze_self, axis, **kwargs):
+        """Returns the sum of each numerical column or row.
 
         Return:
-            Pandas series with the product of each numerical column or row.
+            Pandas series with the sum of each numerical column or row.
         """
-        return DataFrameDefault.register(pandas.DataFrame.prod)(self, **kwargs)
+        # TODO: rework to original implementation after pandas issue #41074 resolves if possible.
+        def map_func(df, **kwargs):
```
Comment (vnlitvinov) [inline modin/backends/base/query_compiler.py:583]: pls add one-line summary docstring

File: modin/backends/base/query_compiler.py [hunk 2]
```diff
@@ -573,21 +573,57 @@ def min(self, **kwargs):
...
+    def prod(self, squeeze_self, axis, **kwargs):
+        """Returns the product of each numerical column or row.
 
         Return:
-            Pandas series with the sum of each numerical column or row.
+            Pandas series with the product of each numerical column or row.
         """
-        return DataFrameDefault.register(pandas.DataFrame.sum)(self, **kwargs)
+        # TODO: rework to original implementation after pandas issue #41074 resolves if possible.
+        def map_func(df, **kwargs):
```
Comment (vnlitvinov) [inline modin/backends/base/query_compiler.py:609]: pls add one-line docstring

File: modin/backends/base/query_compiler.py [hunk 3]
```diff
@@ -573,21 +573,57 @@ def min(self, **kwargs):
         """
         return DataFrameDefault.register(pandas.DataFrame.min)(self, **kwargs)
 
-    def prod(self, **kwargs):
-        """Returns the product of each numerical column or row.
+    def sum(self, squeeze_self, axis, **kwargs):
+        """Returns the sum of each numerical column or row.
```
Comment (vnlitvinov) [inline modin/backends/base/query_compiler.py:577]: what's the reason for swapping the functions?

**Review by vnlitvinov — DISMISSED** [review]
LGTM with some documentation notes

File: modin/backends/base/query_compiler.py [hunk 4]
```diff
@@ -573,21 +573,57 @@ def min(self, **kwargs):
...
+                    if axis:
+                        return pandas.DataFrame(
+                            [result], index=["__reduced__"], columns=["__reduced__"]
+                        )
+                    else:
+                        return pandas.Series([result], index=[df.columns[0]])
+                else:
+                    return result
+            else:
+                return df.sum(**kwargs)
```
Comment (vnlitvinov) [inline modin/backends/base/query_compiler.py:596]: I see functions like this in 6 places (`sum` and `prod` here, and `sum`/`prod`/`sum_min_count`/`prod_min_count` in pandas query compiler), can we reduce the amount of copy-pasteness somehow?

File: modin/backends/base/query_compiler.py (hunk 3 above)
Comment (YarShev) [reply modin/backends/base/query_compiler.py:577]: There is no any reason.

Comment (vnlitvinov) [reply modin/backends/base/query_compiler.py:577]: in that case pls swap them back to reduce the diff :)

File: modin/backends/base/query_compiler.py (hunk 4 above)
Comment (YarShev) [reply modin/backends/base/query_compiler.py:596]: All our map functions look similarly. The difference is that we register different functions.

File: modin/backends/base/query_compiler.py (hunk 1 above)
Comment (YarShev) [reply modin/backends/base/query_compiler.py:583]: added

File: modin/backends/base/query_compiler.py (hunk 2 above)
Comment (YarShev) [reply modin/backends/base/query_compiler.py:610]: added

File: modin/backends/base/query_compiler.py (hunk 3 above)
Comment (YarShev) [reply modin/backends/base/query_compiler.py]: swapped

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @YarShev

**Pull request merged**